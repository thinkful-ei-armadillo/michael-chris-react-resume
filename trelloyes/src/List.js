import React from 'react';
import Card from './Card'
import App from './App.js';
import STORE from './Store.js';
import './List.css';


function List (props){
    
    return (
        <section className="List">
          <header>
            <h2>{props.header}</h2>
          </header>
          <div className="List-cards">
          </div>
        </section>
    )
}

export default List;