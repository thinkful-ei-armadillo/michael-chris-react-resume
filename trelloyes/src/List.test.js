import React from 'react'; 
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import List from './List.js';

describe('List component', () => {
    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<List key={1} header={'First List'} 
            cards={[{id: '0', title: 'First card', content: 'lorem ipsum' }, 
            { id: '1', title: 'First card', content: 'lorem ipsum' }]}/>, div)
        ReactDOM.unmountComponentAtNode(div);
    });

    it('renders the UI as expected', () => {
        const tree = renderer
        .create(<List key={1} header={'First List'} 
        cards={[{id: '0', title: 'First card', content: 'lorem ipsum' }, 
        { id: '1', title: 'First card', content: 'lorem ipsum' }]}/>)
        .toJSON();
        expect(tree).toMatchSnapshot();  
    });
}); 