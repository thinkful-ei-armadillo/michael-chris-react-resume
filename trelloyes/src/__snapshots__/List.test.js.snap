// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List component renders the UI as expected 1`] = `
<section
  className="List"
>
  <header
    className="List-header"
  >
    <h2>
      First List
    </h2>
  </header>
  <div
    className="List-cards"
  >
    <div
      className="Card"
    >
      <h3>
        First card
      </h3>
      <p>
        lorem ipsum
      </p>
    </div>
    <div
      className="Card"
    >
      <h3>
        First card
      </h3>
      <p>
        lorem ipsum
      </p>
    </div>
  </div>
</section>
`;
